cmd/kubeadm/app/phases/upgrade/health.go

```diff
@@ -141,6 +141,7 @@ func createJob(client clientset.Interface, cfg *kubeadmapi.ClusterConfiguration)
 		Spec: batchv1.JobSpec{
 			BackoffLimit:            ptr.To[int32](0),
 			TTLSecondsAfterFinished: ptr.To[int32](int32(timeout.Seconds()) + 5), // Make sure it's more than 'timeout'.
+			ActiveDeadlineSeconds:   ptr.To[int64](int64(timeout.Seconds())),
```

Comment:
just to confirm:
https://github.com/kubernetes/kubernetes/blob/a7702cb47babcff60518b510293dbb41ba8eab35/staging/src/k8s.io/apiserver/pkg/apis/example/v1/types.go#L127-L131

this means that if there is an error such as "image cannot be pulled" the job will fail?
that would be the goal.

we don't want the job to succeed because of the Deadline field here. ideally it should always fail if this timeout is reached,

another Q should it be a value between timeout and timeout +5? is it OK to match the value to the timeout in the poll?

Reply:
> this means that if there is an error such as "image cannot be pulled" the job will fail?

Yes and it should always fail if this timeout is reached.

For more details, after reaching `ActiveDeadlineSeconds` since the start timestamp, the pod fails and the job will have the following condition, marking it for garbage collection thanks to `TTLSecondsAfterFinished`, if the image is not available:
```
status:
  conditions:
  - lastProbeTime: xxx + 20 + something
    lastTransitionTime: xxx + 20
    message: Job was active longer than specified deadline
    reason: DeadlineExceeded
    status: "True"
    type: Failed
  failed: 1
  ready: 0
  startTime: xxx
  terminating: 0
  uncountedTerminatedPods: {}
```

Notice the lastTransitionTime is 20 seconds after the startTime with activeDeadlineSeconds set to 20.

In a success case, the type Complete will make the kubeadm to mark this preflight as pass:
```
status:
  completionTime: xxx + a few sec
  conditions:
  - lastProbeTime: xxx +. a few sec
    lastTransitionTime: xxx + a few sec
    status: "True"
    type: Complete
  ready: 0
  startTime: xxx
  succeeded: 1
  terminating: 0
  uncountedTerminatedPods: {}
```

> another Q should it be a value between timeout and timeout +5? is it OK to match the value to the timeout of the poll?

Let me make this change quickly (also consolidate into one shared var.